src/main/java/org/zalando/nakadi/controller/EventPublishingController.java

```diff
@@ -82,7 +83,7 @@ private ResponseEntity postEventInternal(final String eventTypeName,
             final JSONArray eventsAsJsonObjects = new JSONArray(eventsAsString);
 
             final int eventCount = eventsAsJsonObjects.length();
-            eventTypeMetrics.reportSizing(eventCount, eventsAsString.length());
+            eventTypeMetrics.reportSizing(eventCount, eventsAsString.getBytes(Charsets.UTF_8).length);
```

Comment:
But I actually think that to fix it will be quite easy - we just need to move this line down below "publisher.publish...."
And I think this fix will fit pretty good to this ticket.